sources/packages/api/src/services/education-program/education-program.service.ts

```diff
@@ -40,6 +41,46 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
       .getOne();
   }
 
+  /**
+   * Gets a program details for a student
+   * This returns only a subset of the educationProgram
+   * id in the query.
+   * @param programId Program id.
+   * @returns program
+   */
+  async getStudentEducationProgram(
+    programId: number,
+  ): Promise<StudentEducationProgramSummary> {
+    const studentEducationProgram = await this.repo
+      .createQueryBuilder("programs")
+      .select([
+        "programs.id",
+        "programs.name",
+        "programs.description",
+        "programs.credentialType",
+        "programs.credentialTypeOther",
+        "programs.deliveredOnSite",
+        "programs.deliveredOnline",
+      ])
+      .where("programs.id = :programId", { programId })
+      .getOne();
+
```

Comment:
IMO, the query should be an individual function that returns the repo as promise, and the DTO response can be created inside the controller or as a sep function, then only we could reuse the query right? if the service is specifically for a DTO, then we cant reuse it right? thoughts @guru-aot  @andrewsignori-aot  @dheepak-aot

Reply:
I feel, we should restrict the call in the service level as the student is not authorized to see other columns like the institution. Else we would have used the getInstitutionProgram method of the same service and would have created a separate controller reusing the service.

Reply:
As @ann-aot mentioned, I believe that we has this conversation before and we had a meeting about it. Is there a reason to not follow it?

Reply:
I might have missed it, I think I followed what we discussed. If I have missed something, please explain or we can have a call.

Reply:
Please follow what @ann-aot mentioned or organize a call and we can talk about it.